### PR TITLE
EFI-175 - Remove Trusted Server from the app

### PR DESCRIPTION
**Pull request by martinb89**

### CODE REVIEW

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdAccount/CdAccount+Trusted.swift [hunk 1]
```diff
@@ -13,6 +13,6 @@ extension CdAccount {
         guard let imapServer = server(type: .imap) else {
             return false
         }
-        return imapServer.manuallyTrusted || imapServer.automaticallyTrusted
+        return imapServer.automaticallyTrusted
```
Comment (ghost) [inline SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdAccount/CdAccount+Trusted.swift]: This "trust automatically" is still an option?

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift [hunk 2]
```diff
@@ -231,8 +231,7 @@ extension CdMessage {
             return false
         }
         let accountHasBeenCreatedInLocalNetwork = imapServer.automaticallyTrusted
-        let userDecidedToTrustServer = imapServer.manuallyTrusted
-        return accountHasBeenCreatedInLocalNetwork || userDecidedToTrustServer
+        return accountHasBeenCreatedInLocalNetwork
```
Comment (ghost) [inline SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift]: This still trusts the server if it's a local one. Wanted?

**Review by ghost — CHANGES_REQUESTED** [review]

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift (hunk 2 above)
Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift]: I asked Roberto. It's his decision.

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdAccount/CdAccount+Trusted.swift (hunk 1 above)
Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdAccount/CdAccount+Trusted.swift]: I asked Roberto. It's his decision.

Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdAccount/CdAccount+Trusted.swift]: He said that can be removed only if it doesn't affect the way the app works. I don't know what it does, and I'm not sure what is this for, so I'll keep it.

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift (hunk 2 above)
Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift]: As above, we'll keep it.

**Comment by ghost** [comment]
Then you should probably take a look at QualifyServerIsLocalService. My point is that if, for whatever reason, a server's IP address was identified as local, you still may end up with a trusted server. So `automaticallyTrusted` should go away, to make sure.

Trusted server is a very sensitive topic. If it should go away, it should go away completely, at least in the sense that no account or server can be ever flagged as trusted ever again.

File: SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift (hunk 2 above)
Comment (ghost) [reply SubProjects/MessageModel/MessageModel/MessageModel/CdObjectExtensions/CdMessage/CdMessage+pEp.swift]: Also must be unconditionally false.

File: SubProjects/MessageModel/MessageModel/MessageModel/Services/DecryptService/DecryptMessageOperation.swift [hunk 3]
```diff
@@ -251,10 +251,7 @@ extension DecryptMessageOperation {
 
         let flagNeedsReupload = decryptFlags?.needsReupload ?? false
 
-        if cdMessage.isOnTrustedServer {
-            // Reupload plaintext message for trusted server
-            handleReUploadForTrustedServer(decryptedMessage: cdMessage, originalRating: rating)
-        } else if flagNeedsReupload {
+        if flagNeedsReupload {
```
Comment (ghost) [inline SubProjects/MessageModel/MessageModel/MessageModel/Services/DecryptService/DecryptMessageOperation.swift:254]: We need a separate ticket for verifying this part, I'd say. This is exactly the sort of code that can leak information. Can you please create it? With trusted server gone, there is never a reason to reupload a message right after decryption.

**Review by ghost — APPROVED** [review]
Looks good to, but we need a follow-up ticket for removing code that may still leak mails to the server, just in case (I commented the line).

**Pull request merged**